### PR TITLE
Microsoft sql update

### PR DESCRIPTION
**Pull request by kamarja**

### CODE REVIEW

File: templates/Template_2_SQL_AlwaysOn.template [hunk 1]
```diff
@@ -136,9 +247,85 @@
         "AppServerWaitHandle": {
             "Type": "AWS::CloudFormation::WaitConditionHandle"
         },
+        "AppServerRole": {
```
Comment (santiagocardenas) [inline templates/Template_2_SQL_AlwaysOn.template]: What is this name? AppServerRole?

File: templates/Template_2_SQL_AlwaysOn.template [hunk 2]
```diff
@@ -136,9 +247,85 @@
...
+                                ]
+                            },
+                            "Effect": "Allow"
+                        }
+                    ],
+                    "Version": "2012-10-17"
+                }
+            }
+        },
+        "AppServerProfile": {
```
Comment (santiagocardenas) [inline templates/Template_2_SQL_AlwaysOn.template]: What is this name AppServer? Sounds like copy/paste from SharePoint?

File: templates/sql.template [hunk 3]
```diff
@@ -21,9 +21,9 @@
                         "default": "Amazon EC2 Configuration"
                     },
                     "Parameters": [
-                        "KeyPairName",
+                        "DedicatedHostAMI",
```
Comment (santiagocardenas) [inline templates/sql.template]: Don't change order here.

File: templates/sql.template [hunk 4]
```diff
@@ -63,6 +63,7 @@
                         "default": "Failover Cluster Configuration"
                     },
                     "Parameters": [
+                        "DedicatedHostIDNode3",
```
Comment (santiagocardenas) [inline templates/sql.template]: Don't change order here.

File: templates/Template_2_SQL_AlwaysOn.template [hunk 5]
```diff
@@ -1,10 +1,113 @@
 {
     "AWSTemplateFormatVersion": "2010-09-09",
     "Description": "(0004) - This template creates and installs a Windows Server 2012 based application server for testing purposes into a subnet inside a VPC. **WARNING** This template creates Amazon EC2 Windows instance and related resources. You will be billed for the AWS resources used if you create a stack from this template.",
+    "Metadata": {
```
Comment (santiagocardenas) [inline templates/Template_2_SQL_AlwaysOn.template]: I'd recommend not changing this template as they're deprecated.

File: templates/Template_1_SQL_AlwaysOn.template [hunk 6]
```diff
@@ -1,64 +1,213 @@
...
-        },
-        "WSFCNode2InstanceType": {
-            "Description": "Amazon EC2 instance type for the 1st WSFC Node",
-            "Type": "String",
-            "Default": "r3.2xlarge",
-            "AllowedValues": [
-                "r3.xlarge",
-                "r3.2xlarge",
-                "r3.4xlarge"
+    "Metadata": {
```
Comment (santiagocardenas) [inline templates/Template_1_SQL_AlwaysOn.template]: I'd recommend not changing this template as they're deprecated.

**Comment by kamarja** [comment]
It's just a role for the AppServer instance. I will fix the order in the Params.

**Comment by santiagocardenas** [comment]
What App Server instance?

**Comment by kamarja** [comment]
This App Server instance - https://github.com/aws-quickstart/quickstart-microsoft-sql/blob/master/templates/Template_2_SQL_AlwaysOn.template#L139

**Comment by santiagocardenas** [comment]
Ahh I see. That template is so old, it's more than deprecated. It's completely no longer used.

**Comment by kamarja** [comment]
Ok, I am going to roll back the changes in those 2 templates.

**Comment by kamarja** [comment]
Hey, wait a minute. Template_1_SQL_AlwaysOn.template is still being used by Sharepoint...

**Pull request merged**